src/main/java/bio/terra/service/filedata/azure/AzureSynapsePdao.java

```diff
@@ -1041,23 +1032,31 @@ public List<Map<String, Optional<Object>>> getSnapshotTableData(
     final String sql =
         new ST(queryFromDatasourceTemplate)
             .add("columns", columns.stream().map(Column::getName).toList())
-            .add("datasource", getDataSourceName(snapshot, userRequest.getEmail()))
+            .add("datasource", datasetSourceName)
+            .add("parquetFileLocation", parquetFileLocation)
```

Comment:
Haha I made literally the same change in my PR :)

Reply:
Nice!